apps/x509.c

```diff
@@ -576,7 +578,7 @@ int x509_main(int argc, char **argv)
             BIO_printf(bio_err, "error unpacking public key\n");
             goto end;
         }
-        i = X509_REQ_verify(req, pkey);
+        i = do_X509_REQ_verify(req, pkey, sigopts);
```

Comment:
This is not right from my understandings. Users can only specify one `distid` in the sigopts, but the ID used when signing the CSR may not be the same as the one that is going to use when signing the real certificate from the CSR. Using one sigopts will lose the ability of this.

Reply:
Actually as the commands like `x509`, which deals with both verification and signing (another example is the `ca` command), should provide two separate `opts`: one sigopts and one vfyopts

Reply:
Sure, I can add that

Reply:
Done